### PR TITLE
fix and refactor on NEG annotation handling

### PR DESCRIPTION
**Pull request by freehan**
Refactor NEG annotation handling
Fix a problem where NEG annotation mal format errors are eaten. 

cc: @agau4779

### CODE REVIEW

File: pkg/annotations/service.go [hunk 1]
```diff
@@ -138,59 +154,28 @@ func (svc *Service) ApplicationProtocols() (map[string]AppProtocol, error) {
...
-		return false
-	}
-	return annotation.Ingress
-}
-
 var (
 	ErrBackendConfigNoneFound         = errors.New("no BackendConfig's found in annotation")
 	ErrBackendConfigInvalidJSON       = errors.New("BackendConfig annotation is invalid json")
 	ErrBackendConfigAnnotationMissing = errors.New("BackendConfig annotation is missing")
+	NEGAnnotationErrorFmt             = "NEG annotation is invalid: %v"
```
Comment (rramkumar1) [inline pkg/annotations/service.go]: I don't think its worth surfacing the actual formatting error to users. They can easily figure out what's wrong by throwing it into a json parser. That way, in your tests, you don't have to check for a specific error string.

File: pkg/controller/translator/translator.go [hunk 2]
```diff
@@ -79,7 +79,11 @@ func (t *Translator) getServicePort(id utils.ServicePortID) (*utils.ServicePort,
 		return nil, errors.ErrSvcPortNotFound{ServicePortID: id}
 	}
 
-	negEnabled := annotations.FromService(svc).NEGEnabled()
+	var negEnabled bool
+	ok, negAnnotation, err := annotations.FromService(svc).GetNEGAnnotation()
```
Comment (rramkumar1) [inline pkg/controller/translator/translator.go]: What if there is an error? Where is that handled?

**Comment by freehan** [comment]
Fixed. PTAL

**Pull request merged**